pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -514,43 +514,13 @@ func (a *index) lookupService(key string) *model.ServiceInfo {
 
 // All return all known workloads. Result is un-ordered
 func (a *index) All() []model.AddressInfo {
-	res := dedupeWorkloads(a.workloads.List())
+	res := slices.Map(a.workloads.List(), modelWorkloadToAddressInfo)
 	for _, s := range a.services.List() {
 		res = append(res, s.AsAddress)
 	}
 	return res
 }
 
-func dedupeWorkloads(workloads []model.WorkloadInfo) []model.AddressInfo {
-	if len(workloads) <= 1 {
-		return slices.Map(workloads, modelWorkloadToAddressInfo)
-	}
-	res := []model.AddressInfo{}
-	seenAddresses := sets.New[netip.Addr]()
-	for _, wl := range workloads {
-		write := true
-		// HostNetwork mode is expected to have overlapping IPs, and tells the data plane to avoid relying on the IP as a unique
-		// identifier.
-		// For anything else, exclude duplicates.
-		if wl.Workload.NetworkMode != workloadapi.NetworkMode_HOST_NETWORK {
```

Comment:
Now where do you check the host network?

Reply:
nowhere The check was saying "skip the dedupe for host network". Now we skip the dedupe for everything so we don't need a special case. 

Note ztunnel still has special case for the host net

Reply:
IC